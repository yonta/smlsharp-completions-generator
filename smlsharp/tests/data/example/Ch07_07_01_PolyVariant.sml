val myCPoint = fn M => #CPoint M {x = 1.0, y = 1.0};
val myPPoint = fn M => #PPoint M {r = 1.41421356237, theta = 45.0};

val distance = 
{
  CPoint = fn {x, y, ...} => Real.Math.sqrt (x * x + y * y),
  PPoint = fn {r, theta, ...} => r
};

myCPoint distance;
myPPoint distance;
