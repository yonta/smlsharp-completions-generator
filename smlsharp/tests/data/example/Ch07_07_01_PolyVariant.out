val myCPoint = fn : ['a#{CPoint: {x: real, y: real} -> 'b}, 'b. 'a -> 'b]

val myPPoint = fn : ['a#{PPoint: {r: real, theta: real} -> 'b}, 'b. 'a -> 'b]

val distance =
  {CPoint = fn, PPoint = fn}
  : {CPoint: ['a#{x: real, y: real}. 'a -> real],
     PPoint: ['a#{r: 'b, theta: 'c}, 'b, 'c. 'a -> 'b]}

val it = 1.41421356237 : real

val it = 1.41421356237 : real
